snuba/clickhouse/columns.py

```diff
@@ -93,19 +97,19 @@ def __eq__(self, other):
             and self.inner_type == other.inner_type
         )
 
-    def for_schema(self):
+    def for_schema(self) -> str:
         return "{} MATERIALIZED {}".format(
             self.inner_type.for_schema(), self.expression,
         )
 
 
 class WithDefault(ColumnType):
-    def __init__(self, inner_type, default):
+    def __init__(self, inner_type: ColumnType, default) -> None:
         self.inner_type = inner_type
-        self.default = default
+        self.default = default  # XXX: this is problematic for typing
```

Comment:
This is only really going to be viable if the column type itself is generic on it's data type. For example,

- `ColumnType` would `ColumnType[T]`, 
- `Float(ColumnType)` would become `Float(ColumnType[float])`, 
- `WithDefault(ColumnType)` would become `WithDefault(ColumnType[T])`

Whether this is worth it or not is a different question that I intentionally avoided answering here. 🙂